### PR TITLE
Introduce Add user to project and update user in project

### LINKED ISSUE #484: Ability to manage users in project
- Add user to project
- Enable/Disable user in project

### PR DESCRIPTION
**Pull request by Tantalor93**
close #484

### CODE REVIEW

File: src/main/java/com/gooddata/project/ProjectService.java [hunk 1]
```diff
@@ -365,4 +370,76 @@ public CreatedInvitations sendInvitations(final Project project, final Invitatio
             throw new GoodDataException("Unable to invite " + emails + " to project " + project.getId(), e);
         }
     }
+
+    /**
+     * get user in project
+     *
+     * @param project where to find
+     * @param account which user to find
+     * @return User representation in project
```
Comment (martiner) [inline src/main/java/com/gooddata/project/ProjectService.java]: document throws

File: src/main/java/com/gooddata/project/ProjectService.java [hunk 2]
```diff
@@ -365,4 +370,76 @@ public CreatedInvitations sendInvitations(final Project project, final Invitatio
...
+        notNull(users, "users");
+        notEmpty(project.getId(), "project.id");
+
+        doPostProjectUsersUpdate(project, users);
+    }
+
+    private void doPostProjectUsersUpdate(final Project project, final User... users) {
+        final URI usersUri = getUsersUri(project);
+
+        final ProjectUsersUpdateResult projectUsersUpdateResult = restTemplate.postForObject(usersUri, new Users(users), ProjectUsersUpdateResult.class);
```
Comment (martiner) [inline src/main/java/com/gooddata/project/ProjectService.java]: catch rest exception

File: src/main/java/com/gooddata/project/UsersSerializer.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import com.fasterxml.jackson.core.ObjectCodec;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+
+import java.io.IOException;
+
+/**
+ * Serializer of {@link Users}
+ */
+public class UsersSerializer extends JsonSerializer<Users> {
```
Comment (martiner) [inline src/main/java/com/gooddata/project/UsersSerializer.java]: package private

File: src/main/java/com/gooddata/project/ProjectService.java (hunk 1 above)
Comment (Tantalor93) [inline src/main/java/com/gooddata/project/ProjectService.java]: done

File: src/main/java/com/gooddata/project/ProjectService.java (hunk 2 above)
Comment (Tantalor93) [inline src/main/java/com/gooddata/project/ProjectService.java]: done

File: src/main/java/com/gooddata/project/UsersSerializer.java (hunk 3 above)
Comment (Tantalor93) [inline src/main/java/com/gooddata/project/UsersSerializer.java]: done

**Comment by Tantalor93** [comment]
ok to test.

File: src/main/java/com/gooddata/project/ProjectService.java [hunk 4]
```diff
@@ -369,4 +375,81 @@ public CreatedInvitations sendInvitations(final Project project, final Invitatio
...
+
+    /** Add user in to the project
+     *
+     * @param project where to add user
+     * @param account to be added
+     * @param userRoles list of user roles
+     * @return user added to the project
+     * @throws ProjectUsersUpdateException in case of failure
+     */
+    public User addUserToProject(final Project project, final Account account, final List<Role> userRoles) {
```
Comment (martiner) [inline src/main/java/com/gooddata/project/ProjectService.java]: roles as vararg

File: src/test/java/com/gooddata/AbstractGoodDataAT.java [hunk 5]
```diff
@@ -72,4 +75,11 @@ public static void removeProjectAndLogout() {
         }
         gd.logout();
     }
+
+    @AfterSuite
+    public void removeAccount() {
```
Comment (martiner) [inline src/test/java/com/gooddata/AbstractGoodDataAT.java]: not sure whether this is a good idea. firstly - you removed a test. secondly this after method is executed always even when tests are not run by domain admin

File: src/main/java/com/gooddata/project/ProjectService.java (hunk 4 above)
Comment (Tantalor93) [inline src/main/java/com/gooddata/project/ProjectService.java]: done

File: src/test/java/com/gooddata/AbstractGoodDataAT.java (hunk 5 above)
Comment (Tantalor93) [inline src/test/java/com/gooddata/AbstractGoodDataAT.java]: reverted, another account created in ProjectServiceAT

**Pull request merged**